### PR TITLE
Add ability to enabled/disable components

### PR DESCRIPTION
**Pull request by rzezeski**
This is actually a rename of the `noop_flag` API.
- Add a `component()` type that enumerates the components that may be
  enabled or disabled.
- Add functions to the `yokozuna` module to enabled/disable components
  and check if a component is enabled.
- By default a component is considered enabled.
- Honor the search enabled flag in the protobuff search path.
- Add better code documentation and specs.

### CODE REVIEW

File: include/yokozuna.hrl [hunk 1]
```diff
@@ -77,6 +77,23 @@
                    | {siblings, binary()}
                    | {'query', binary()}.
 
+%% @doc The `component()' type reprsents components that may be
```
Comment (coderoshi) [inline include/yokozuna.hrl]: represents

File: src/yz_kv.erl [hunk 2]
```diff
@@ -110,7 +110,7 @@ get_md_entry(MD, Key) ->
     yz_misc:dict_get(Key, MD, none).
 
 index(Obj, Reason, VNodeState) ->
-    case yokozuna:noop_flag(index) orelse not ?YZ_ENABLED of
+    case yokozuna:is_enabled(index) orelse not ?YZ_ENABLED of
```
Comment (coderoshi) [inline src/yz_kv.erl]: this should check if "not enabled" (=noop), or better yet:

case yokozuna:is_enabled(index) andalso ?YZ_ENABLED of
  false -> ok;
  true -> %% do work...

**Comment by rzezeski** [comment]
How about now @coderoshi?

**Comment by coderoshi** [comment]
+1, though https://github.com/basho/yokozuna/pull/147/files#L2R113 could be clarified

**Pull request merged**